### PR TITLE
Alerting: Update to alerting  20230203015918-0e4e2675d7aa (after refactoring)

### PR DESCRIPTION
**Pull request by yuri-tceretian**
**What is this feature?**
This is the second part of the refactoring inhttps://github.com/grafana/alerting/pull/49. It updates to the recent alerting repository and updates usages to point to new packages.
Also, it removes the notifiers factory because it was moved to the alerting repo

### CODE REVIEW

File: pkg/services/ngalert/api/api_alertmanager.go [hunk 1]
```diff
@@ -10,7 +10,7 @@ import (
 	"time"
 
 	"github.com/go-openapi/strfmt"
-	"github.com/grafana/alerting/alerting"
+	"github.com/grafana/alerting/notify"
```
Comment (gotjosh) [inline pkg/services/ngalert/api/api_alertmanager.go:13]: How do you feel about using the prefix `alerting_` for all `grafana/alerting` packages?

Right now, the Alertmanager is _still_ imported directly in some areas (e.g. configuration), so it's a bit confusing to know if this is the upstream notify package or the alerting package.

While we do have plans to eventually remove the direct dependency of the Alertmanager until we do I say we avoid shooting ourselves in the foot and simply use `alerting_notify`, `alerting_receivers` et al.

Once the complete removal of the direct Alertmanager dependency is done - it'll be dead obvious that unless you see a modification to the `go.mod` you aren't importing something related to the Alertmanger directly and thus removes the need for this rename.

File: pkg/services/ngalert/notifier/images.go [hunk 2]
```diff
@@ -4,7 +4,7 @@ import (
 	"context"
 	"errors"
 
-	"github.com/grafana/alerting/alerting/notifier/channels"
+	"github.com/grafana/alerting/images"
```
Comment (gotjosh) [inline pkg/services/ngalert/notifier/images.go:7]: My comment above can be ignore for these packages as they don't have an alertmanager equivalent.

**Review by gotjosh — COMMENTED** [review]
In practice, this looks good to me - can you please see my comment?

File: pkg/services/ngalert/api/api_alertmanager.go (hunk 1 above)
Comment (yuri-tceretian) [reply pkg/services/ngalert/api/api_alertmanager.go]: I decided to use camel case as we already use `alertingModels` everywhere and GoLand's default style settings do not seem to like snake case.

**Review by gotjosh — APPROVED** [review]
LGTM

**Pull request merged**